src/arc_utilities/filesystem_utils.py

```diff
@@ -23,14 +23,14 @@ def mkdir_and_ask(path, parents: bool, yes: Optional[bool] = False):
     return True
 
 
-def get_all_subfolders(args):
-    all_subfolders = []
-    for results_dir in args.results_dirs:
-        subfolders = results_dir.iterdir()
-        for subfolder in subfolders:
-            if subfolder.is_dir():
-                all_subfolders.append(subfolder)
-    return all_subfolders
+def get_all_subdirs(directories: Iterable[pathlib.Path]):
+    all_subdirs = []
+    for results_dir in directories:
+        subdirs = results_dir.iterdir()
```

Comment:
nit: More pythonic, less verbose, same logic.

If you wanted, you could include the outer for loop in the list comprehension as well

`all_subdirs += [file for file in results_dir.iterdir() if file.is_dir()]`